elyra/pipeline/processor_airflow.py

```diff
@@ -73,26 +73,15 @@ def process(self, pipeline):
 
             self.log.debug("Uploading pipeline file: %s", pipeline_filepath)
 
-            github_client = Github(login_or_token=github_repo_token)
-            try:
-                # Upload to github
-                repo = github_client.get_repo(github_repo)
+            github_client = GithubClient(server_url=runtime_configuration.metadata.get('github_api_endpoint'),
+                                         token=github_repo_token,
+                                         repo=github_repo,
+                                         branch=github_branch)
 
-                with open(pipeline_filepath) as input_file:
-                    content = input_file.read()
-
-                    repo.create_file(path=pipeline_name + ".py",
-                                     message="Pushed DAG " + pipeline_name,
-                                     content=content,
-                                     branch=github_branch)
-
-                self.log.info('Pipeline successfully added to the Airflow git queue')
-                self.log.info('Waiting for Airflow Scheduler to process and start the pipeline')
-
-            except GithubException as e:
-                self.log.debug('Error adding pipeline to Airflow git queue: ' + e)
-                raise RuntimeError('Error adding pipeline to Airflow git queue: ', e)
+            github_client.upload_dag(pipeline_name=pipeline_name,
+                                     pipeline_filepath=pipeline_filepath)
 
+            self.log.info('Waiting for Airflow Scheduler to process and start the pipeline')
             self.log_pipeline_info(pipeline_name,
                                    f"pipeline pushed to git: https://github.com/{github_repo}/tree/{github_branch}",
```

Comment:
With the added support for a user-provided server URL we can't use `https://github.com/` anymore

Reply:
updated with function in git.py util to generate a url link